### ISSUE TITLE
mDNS tiles don't always show after refresh

### ISSUE THREAD

**Issue by wilsonpage** [mozilla-magnet/magnet-client#197]
Steps:
1. Discover mdns tile
2. Pull refresh

Expected: mdns tile should reappear.
Actual: mdns tile take very long tome to reappear

---

This is likely because mdns advertisements are infrequent and have a 'found' and 'lost' callbacks. Therefore as the scanner service is long lived, the app may have missed the advertisement.

To mitigate this, the scanner service should keep a list of found items and pass them back to the client as a result to `.start()`. That way the client doesn't depend on another 'found' mdns broadcast to discover adverts.

This is also relevant to the `NotificationService`, which may not be aware of mdns adverts that were received by the `ScannerService` before it called `.start()`.

In summary a client should say: "hey service, give me all the stuff you know about already, and let me know when new stuff is found".
